pkg/sidecar/sidecar.go

```diff
@@ -66,14 +73,56 @@ func New(configPath string, log logrus.FieldLogger) (*Sidecar, error) {
 	}, nil
 }
 
+// RunDaemon starts the main loop
+// Starts the workload API client to listen for new SVID updates
+// When a new SVID is received on the updateChan, the SVID certificates
+// are stored in disk and a restart signal is sent to the proxy's process
+func (s *Sidecar) RunDaemon(ctx context.Context) error {
+	var wg sync.WaitGroup
+
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, s.getWorkloadAPIAdress())
+			if err != nil && status.Code(err) != codes.Canceled {
+				s.config.Log.Errorf("Error watching X.509 context: %v", err)
+			}
+		}()
+	}
+
+	if s.config.JWTBundleFilename != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := workloadapi.WatchJWTBundles(ctx, &JWTBundlesWatcher{sidecar: s}, s.getWorkloadAPIAdress())
+			if err != nil && status.Code(err) != codes.Canceled {
+				s.config.Log.Errorf("Error watching JWT bundle updates: %v", err)
+			}
+		}()
+	}
+
+	if s.config.JWTSvidFilename != "" && s.config.JWTAudience != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			s.updateJWTSVID(ctx, s.getWorkloadAPIAdress())
+		}()
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
 // CertReadyChan returns a channel to know when the certificates are ready
 func (s *Sidecar) CertReadyChan() <-chan struct{} {
 	return s.certReadyChan
 }
 
 // updateCertificates Updates the certificates stored in disk and signal the Process to restart
 func (s *Sidecar) updateCertificates(svidResponse *workloadapi.X509Context) {
-	s.config.Log.Info("Updating certificates")
+	s.config.Log.Info("Updating X.509 certificates")
```

Comment:
```suggestion
	s.config.Log.Debug("Updating X.509 certificates")
```

And add after the successful `dumpBundles` operation:

```
       s.config.Log.Info("X.509 certificates updated")
```